component/telemetry.go

```diff
@@ -16,6 +16,7 @@ package component // import "go.opentelemetry.io/collector/component"
 
 import (
 	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
```

Comment:
That's a fair concern, i don't know that we expose the otel sdk anywhere else in the collector API... I'm ok with using `pcommon.Resource` here

Reply:
sdk's resource is used to initialize the telemetry [here](https://github.com/open-telemetry/opentelemetry-collector/pull/7531/files#diff-52da5566fa270d6a253f7a301b66f4a1dbda7c2c42f3781f8357efe7f08dff10L210) which is why I went with this resource to start with – it allows us to only initialize the sdk resource once and pass it down. Would it be straightforward to convert `pcommon.Resource` to the sdk's resource for initializing the otel sdk?

Reply:
I'd agree we should avoid making a dependency on the Go SDK in the Collector API since Collector distributions should be free to choose their own SDK. I don't see any Collector APIs where the SDK is currently used, either.

Converting from `pcommon.Resource` to an SDK resource should be fairly easy by just iterating through the underlying map, e.g. through `pResource.Attributes().Range(...)`.

Reply:
@evan-bradley @bogdandrutu are there any examples of initializing a `pcommon.Resource` every one i've found so far are using this method, which has a note to only use it in tests.
```
// This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
// OR directly access the member if this is embedded in another struct.
func NewResource() Resource {
```

Everything else would require me to directly access the internal package which is forbidden. What should i be doing here?
<img width="1103" alt="Screen Shot 2023-04-26 at 3 22 24 PM" src="https://user-images.githubusercontent.com/10070047/234681127-b7cfea54-91de-486e-b165-e942155629d6.png">

Reply:
i ended up using this method as that's the only way to do this as far as i can tell. Please let me know if this is okay, and i'll update the docs.

Reply:
I think the comment should likely be changed. The apparent reason for the comment is that anything constructing pdata should use e.g. `ptrace.NewTraces()` and just use `ResourceSpans().Attributes()` to set attributes. However, that's not possible if the resource attributes are signal-agnostic, as they are here or in some of the places where `pcommon.NewResource` is used, such as in the resource detection processor.

Someone else can feel free to correct me if I've missed something, but I think this usage is okay.

Reply:
oy, looks like changing this comment is going to be a PITA... I'll comment on the usage in buildResource.